ogr/ogrsf_frmts/miramon/mm_gdal_functions.c

```diff
@@ -0,0 +1,2801 @@
+                {
+                    MM_strnzcpy(pMMBDXP->pField[nIField]
+                                    .FieldDescription[MM_DEF_LANGUAGE],
+                                pszString, MM_MAX_LON_DESCRIPCIO_CAMP_DBF);
+                }
+                free_function(pszString);
+            }
+            else
+                *pMMBDXP->pField[nIField].FieldDescription[MM_ENG_LANGUAGE] =
+                    '\0';
+
+            // MM_CAT_LANGUAGE
+            pszDesc = MMReturnValueFromSectionINIFile(pszRelFile, section,
+                                                      "descriptor_cat");
+            if (pszDesc)
+            {
+                pszString = CPLRecode_function(pszDesc, CPL_ENC_ISO8859_1,
+                                               CPL_ENC_UTF8);
+                free_function(pszDesc);
+
+                MM_strnzcpy(
+                    pMMBDXP->pField[nIField].FieldDescription[MM_CAT_LANGUAGE],
+                    pszString, MM_MAX_LON_DESCRIPCIO_CAMP_DBF);
+
+                if (*pMMBDXP->pField[nIField]
+                         .FieldDescription[MM_DEF_LANGUAGE] != '\0')
+                {
+                    MM_strnzcpy(pMMBDXP->pField[nIField]
+                                    .FieldDescription[MM_DEF_LANGUAGE],
+                                pszString, MM_MAX_LON_DESCRIPCIO_CAMP_DBF);
+                }
+
+                free_function(pszString);
+            }
+            else
+                *pMMBDXP->pField[nIField].FieldDescription[MM_CAT_LANGUAGE] =
+                    '\0';
+
+            // MM_SPA_LANGUAGE
+            pszDesc = MMReturnValueFromSectionINIFile(pszRelFile, section,
+                                                      "descriptor_cat");
+            if (pszDesc)
+            {
+                pszString = CPLRecode_function(pszDesc, CPL_ENC_ISO8859_1,
+                                               CPL_ENC_UTF8);
+                free_function(pszDesc);
+
+                MM_strnzcpy(
+                    pMMBDXP->pField[nIField].FieldDescription[MM_SPA_LANGUAGE],
+                    pszString, MM_MAX_LON_DESCRIPCIO_CAMP_DBF);
+
+                if (*pMMBDXP->pField[nIField]
+                         .FieldDescription[MM_DEF_LANGUAGE] != '\0')
+                {
+                    MM_strnzcpy(pMMBDXP->pField[nIField]
+                                    .FieldDescription[MM_DEF_LANGUAGE],
+                                pszString, MM_MAX_LON_DESCRIPCIO_CAMP_DBF);
+                }
+
+                free_function(pszString);
+            }
+            else
+                *pMMBDXP->pField[nIField].FieldDescription[MM_SPA_LANGUAGE] =
+                    '\0';
+        }
+    }
+
+    if (!pMMBDXP->nFields)
+    {
+        if (pMMBDXP->BytesPerRecord)
+            incoherent_record_size = TRUE;
+    }
+    else if (pMMBDXP->pField[pMMBDXP->nFields - 1].BytesPerField +
+                 pMMBDXP->pField[pMMBDXP->nFields - 1].AcumulatedBytes >
```

Comment:
done:
This before the code that can overflow
// To avoid overflow
        if (pMMBDXP->pField[pMMBDXP->nFields - 1].AcumulatedBytes <
            UINT32_MAX - pMMBDXP->pField[pMMBDXP->nFields - 1].BytesPerField)
        {
            free_function(pMMBDXP->pField);
                    pMMBDXP->pField = nullptr;
                    fclose_function(pf);
                    return 1;
        }

Reply:
Hum, no, that should be a ``>`` comparison here. The overflow is when the sum would exceed UINT32_MAX.